### ISSUE TITLE
IDN mentions not being processed

### ISSUE THREAD

**Issue by tigransimonyan** [mastodon/mastodon#16602]
### Expected behaviour

Mention someone from another instance with a unicode domain

### Actual behaviour

Mention not working

### Steps to reproduce the problem

1. Follow `@tigran@xn--y9a6bah4ck.xn--y9a3aq` (@tigran@անքուն.հայ)
2. Open the web interface and write `@tigran`.  Popup will prompt you to choose `@tigran@անքուն.հայ`
3. After selection, the user will not be notified and mentioned. 

### Specifications

macOS Catalina, Firefox 90.0.2, Mastodon v3.4.1

**Comment by tigransimonyan** [comment]
This problem is worse than I described.
When someone mentioned me as "@tigran@անքուն.հայ" the message was sent to "@tigran" in the sender's instance 😔

**Comment by ClearlyClaire** [comment]
It seems like this happens with IDN TLDs exclusively, the regexp expecting the domain part of mentions to end with ascii alphanumeric characters:
```ruby
/(?<=^|[^\/[:word:]])@((#{USERNAME_RE})(?:@[[:word:]\.\-]+[a-z0-9]+)?)/i
```

This is an oversight that should be fixed by #16689 although I'm unsure if the regexp might be too eager and mess up other things.

**Comment by tigransimonyan** [comment]
Many thanks!
